package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,368 @@
+                                      data_selector=data_selector,
+                                      **kwargs)
+
+    def _select_time(self, key: str) -> np.float64:
+        """'Time' is one of the entries in the dict returned by pyedr.
+        The base AuxStep Class uses the time_selector 'Time' to return the
+        time value of each step."""
+        return self._select_data(key)
+
+    def _select_data(self, key: Union[str, None]) -> np.float64:
+        if key is None:
+            return
+        try:
+            return self._data[key]
+        except KeyError:
+            raise KeyError(f"'{key}' is not a key in the data_dict dictionary."
+                           " Check the EDRReader.terms attribute")
+
+
+class EDRReader(base.AuxReader):
+    """ Auxiliary reader to read data from an .edr file.
+
+    Default reader for .edr files. All data from the file will be read and
+    stored on initialisation.
+
+    Parameters
+    ----------
+    filename : str
+        Location of the file containing the auxiliary data.
+    **kwargs
+       Other AuxReader options.
+
+    Attributes
+    ----------
+    _auxdata : pathlib.PosixPath
+        path at which the auxiliary data file is located
+    data_dict : dict
+        dictionary that contains the auxiliary data, mapping the name of the
+        data point to a NumPy array
+    unit_dict : dict
+        dictionary that contains the units of the auxiliary data, mapping the
+        :attr:`data_selector` of the Reader (i.e. the name of the dataset in
+        the EDR file) to its unit.
+    _n_steps : int
+        Number of steps for which auxdata is available
+    terms : list
+        Names of the auxiliary data entries available in `data_dict`
+
+    See Also
+    --------
+    :class:`~MDAnalysis.auxiliary.base.AuxReader`
+    :func:`~MDAnalysis.coordinates.base.add_auxiliary`
+
+    Note
+    ----
+    The file is assumed to be of a size such that reading and storing the full
+    contents is practical.
+    """
+
+    format = "EDR"
+    _Auxstep = EDRStep
+
+    def __init__(self, filename: str, **kwargs):
+        self._auxdata = Path(filename).resolve()
+        self.data_dict = pyedr.edr_to_dict(filename)
+        self.unit_dict = pyedr.get_unit_dictionary(filename)
+        self._convert_units()
+        self._n_steps = len(self.data_dict["Time"])
+        # attribute to communicate found energy terms to user
+        self.terms = list(self.data_dict.keys())
+        super(EDRReader, self).__init__(**kwargs)
+
+    def _convert_units(self):
+        """Called during :func:`__init__` to convert the units found in the EDR
+        file to MDAnalysis base units"""
+        for term, unit in self.unit_dict.items():
+            try:
+                unit_type = units.unit_types[unit]
+            except KeyError:
+                warnings.warn(f"Could not find unit type for unit '{unit}'.")
+                continue  # skip conversion if unit not defined yet
```

Comment:
Document in the higher-level methods and at the class level how units are treated.

Make sure to state which quantities are converted and which are not — this might deserve a warning box. If you create `units.MDANALYSIS_BASE_UNITS` then you can link to it in the docs to simply say, anything in this dict is converted.